database_wrapper.py

```diff
@@ -5,43 +5,221 @@
 
     Typical usage example:
 
-    connection = connect_to_mysql('config.json')
-    bar = foo.FunctionBar()
+    db = NimbusMySQL(config_file="config.json")
+    ents = db.get_entities()
 """
 #!/usr/bin/env python3
 import mysql.connector
+from abc import ABC, abstractmethod
+from typing import Any, Optional
 import json
 
-MYSQL_TYPE = mysql.connector.connection_cext.CMySQLConnection
 
+class UnsupportedDatabaseError(Exception):
+    """Raised when operation tries to connect to an unsupported database type.
+
+    Attributes:
+        message: an explanation of why the expected database is unsupported.
+    """
+
+    def __init__(self, message: str):
+        self.message = message
+
+
+class BadConfigFileError(Exception):
+    """Raised when the config.json file is badly formatter (e.g missing field).
+
+    Attributes:
+        message: an explanation.
+    """
+
+    def __init__(self, message: str):
+        self.message = message
+
+
+        
+        
+"""
+utilities.py
```

Comment:
Should these functions be in a separate file?

Reply:
that is a utilities.py file

Reply:
yes! TODO! 😅